### ISSUE TITLE
Finding a valid point on the NavMesh is inconsistent.

### ISSUE THREAD

**Issue by alcoranpaul** [FlaxEngine/FlaxEngine#2398]
Goal: Move Ai towards Rectangle model.

**Issue description:**

~~I had trouble with moving my model with a collider for an AI enemy to move towards~~.

When my Base, rectangle model that has a red wire sphere, is placed on certain spots, 
`Navigation.FindRandomPointAroundCircle` or `Navigation.FindClosestPoint` returns false. Thus my ai is failing to find a path.

The functions `Navigation.FindRandomPointAroundCircle` or `Navigation.FindClosestPoint` returns false 
even though if I move it slightly around the original area, it will return a valid nav mesh poi

I also had tweaked and tested different radius for the Navigation setting for the Agent.

**Steps to reproduce:**
1. Have a level prepared
2. Have a target for the AI to move towards that has: BoxCollider, Cube Model, and NavModifier.
> -  I have tried with Model as the root and BoxCollider as the root and both yielded same result
3. Implement Simple Move from [tutorial](https://docs.flaxengine.com/manual/navigation/tutorials/path-following.html) or use the following code below on the AI
4. Static flag for the target is full static checked
> - I have tested with only checking Transform flag and unchecking Navigation flag to be working but it dont make sense that `FindRandomPointAroundCircle` or `FindClosestPoint` doesnt return a valid NavMeshPoint when Full Static is checked

```csharp
/// <summary>
    /// The target object to follow.
    /// </summary>
    public Actor MoveToTarget;

 public override void OnStart()
    {
        // See initially if target is reachable
        bool navPoint = Navigation.FindClosestPoint(MoveToTarget.Position, out targetPosition);
        Debug.Log($"HIT: {navPoint} targetPosition: {targetPosition} ");

        if (navPoint)
        {
            BoundingSphere sp2 = new BoundingSphere(targetPosition, 50f);
            DebugDraw.DrawWireSphere(sp2, Color.Blue, 10f);
        }


    }

    /// <inheritdoc />
public override void OnUpdate()
    {
        // Check if any areas are reachable near the target
        GetValidNavMeshCircleRandomPoint(MoveToTarget);
   }

public void GetValidNavMeshCircleRandomPoint(Actor start)
    {
        float rad = 200f;
        BoundingSphere sp = new BoundingSphere(start.Position, rad);
        DebugDraw.DrawWireSphere(sp, Color.Red, 10f);
        bool cloestP = Navigation.FindRandomPointAroundCircle(start.Position, rad, out Vector3 tar);
        Debug.Log($"CircleRandomPoint: {cloestP} - targetPos: {tar}");
        if (cloestP)
        {

            BoundingSphere sp2 = new BoundingSphere(tar, 50f);
            DebugDraw.DrawWireSphere(sp2, Color.Green, 10f);


        }
    }
```

**Files**
- The Green Debug Spheres are Valid Random Points around the Target
- The Red Debug Sphere is the Radius of the Circle for the function `Navigation.FindRandomPointAroundCircle`

Ground Level:
- First Play, Ai has failed to find a valid point
- Second play, Ai has found multiple valid points

https://github.com/FlaxEngine/FlaxEngine/assets/67833935/9a026ffc-9cca-4dc6-9ebe-dd22a6fd9563

Higher Level:

https://github.com/FlaxEngine/FlaxEngine/assets/67833935/14735bdf-a38a-47f5-b104-01e9aeaac47b

Navigation Setting
![image](https://github.com/FlaxEngine/FlaxEngine/assets/67833935/56115c42-9c1a-4d0c-b6aa-e33ed514654a)

**Flax version:**
1.8 + Master

**Comment by Menotdan** [comment]
I suspect this is due to the fact that your agent is labelled as static and thus gets baked into the navmesh which prevents it from moving. As for why it works ever at all, I have no clue.

**Comment by alcoranpaul** [comment]
The problem is that there is no valid nav mesh point being returned by the functions described above. Thus the Ai will fail to find a path towards the target. Sorry if that was not clear!

The AI"s flag are set to None and the Base which is the Rectangle model that has the static flag enabled.

**Comment by Menotdan** [comment]
> The problem is that there is no valid nav mesh point being returned by the functions described above. Thus the Ai will fail to find a path towards the target. Sorry if that was not clear!
> 
> The AI"s flag are set to None and the Base which is the Rectangle model that has the static flag enabled.

Oh. The unclear part wasn't that, I thought you were trying to move the rectangle model, not path to it lol

**Comment by alcoranpaul** [comment]
> Oh. The unclear part wasn't that, I thought you were trying to move the rectangle model, not path to it lol

Oh lol, yeah I wanted my AI to find a path to the rectangle model. 
I assumed that the functions for finding a valid navmesh point would work regardless of the point being on or off the navmesh to still work. 


I think in Unity, it has a function for finding a nearest valid point regardless of being on or off the navmesh.

**Comment by Menotdan** [comment]
Yeah it should work, this sounds like a bug.

**Comment by mafiesto4** [comment]
Fixed in 41ffc16b66d756e82b160b1803a97512e0c3f0f2.
